extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -78,7 +80,15 @@ trait LineageParser {
   private def getSelectColumnLineage(
       named: Seq[NamedExpression]): AttributeMap[AttributeSet] = {
     val exps = named.map {
-      case a: Alias => a.toAttribute -> a.references
+      case exp: Alias =>
+        exp.child match {
```

Comment:
what happens for `select (select count(*) from t0) + 1 as c from t1` ? we should find the inside subquery rather than match the top level.

Reply:
The logical plan for the above SQL:
```
Project [(scalar-subquery#2 [] + 1) AS aa#3L, b#6 AS bb#4]
:  +- Aggregate [count(1) AS count(1)#12L]
:     +- Project
:        +- HiveTableRelation [`default`.`table1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#9, b#10, c#11], Partition Cols: []]
+- HiveTableRelation [`default`.`table1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#5, b#6, c#7], Partition Cols: []]
```
IMO,  is that enough to find the `ScalarSubquery` by `match` here, and re-parse scalar subquery?

Reply:
oh, sorry the `count(*)` may be confused. how about:
`select (select a from t0 limit 1) + 1 as c from t1`